src/PrestaShopBundle/Entity/Repository/CmsPageRepository.php

```diff
@@ -0,0 +1,110 @@
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace PrestaShopBundle\Entity\Repository;
+
+use CMS;
+use Language;
+use PrestaShop\PrestaShop\Core\Domain\CmsPage\CmsPageSettings;
+use PrestaShop\PrestaShop\Core\Domain\CmsPage\Exception\CannotDuplicateCmsPageException;
+use PrestaShop\PrestaShop\Core\Domain\CmsPage\Exception\CmsPageNotFoundException;
+use PrestaShop\PrestaShop\Core\Domain\CmsPage\ValueObject\CmsPageId;
+use PrestaShop\PrestaShop\Core\Repository\AbstractObjectModelRepository;
+use PrestaShop\PrestaShop\Core\Util\String\StringModifierInterface;
+use Symfony\Component\Translation\TranslatorInterface;
+
+class CmsPageRepository extends AbstractObjectModelRepository
+{
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    /**
+     * @var StringModifierInterface
+     */
+    private $stringModifier;
+
+    /**
+     * @param TranslatorInterface $translator
+     * @param StringModifierInterface $stringModifier
+     */
+    public function __construct(
+      TranslatorInterface $translator,
+      StringModifierInterface $stringModifier
+  ) {
+      $this->translator = $translator;
+      $this->stringModifier = $stringModifier;
+  }
+
+    /**
+     * Duplicates a CMS page
+     *
+     * @param CmsPageId $cmsPageId
+     *
+     * @return CmsPageId
+     *
+     * @throws CmsPageNotFoundException
+     * @throws CannotDuplicateCmsPageException
+     */
+    public function duplicate(CmsPageId $cmsPageId): CmsPageId
+    {
+        /** @var Cms $cms */
+        $cms = $this->getObjectModel(
+          $cmsPageId->getValue(),
+          Cms::class,
+          CmsPageNotFoundException::class
+        );
+
+        // Reset object ID, make it inactive and save it
+        $cms->id = null;
+        $cms->meta_title = $this->getNewPageTitles($cms->meta_title);
+	      $cms->active = false;
+        $id = $this->addObjectModel($cms, CannotDuplicateCmsPageException::class);
+
+        return new CmsPageId($id);
+    }
+
+    /**
+     * Provides duplicated cms page title
+     *
+     * @param array<int, string> $oldCmsPageLocalizedTitles
+     *
+     * @return array<int, string>
+     */
+    private function getNewPageTitles(array $oldCmsPageLocalizedTitles): array
```

Comment:
good job introducing the repo :muscle: 
Though about this method - it is same as in product page duplication. So as you mentioned in private, maybe we could think of a way to reuse it in both places?
Another thing Im concerned about is that such logic doesn't belong to a "repository" design pattern, so probably we will end up creating dedicated CmsDuplicator service anyway.

Could we have some AbstractDuplicator service that could then be reused by both Product and Cms to rename these localized titles?
I need help deciding this @jolelievre :pray:

Reply:
I guess it would even be worth having a `LocalizedFieldDuplicator` service instead that is injected in both those services and can be used via a method:

```php
public function duplicateLocalizedValues(array $localizedValues): array
```

And this service is the one in which we inject the translator. This way it can also be unit tested and resued in many other use cases potentially.

Reply:
Done ✅